http/sys_pprof_index_requests.go

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package http
+
+import (
+	"net/http"
+	"net/http/pprof"
+	"runtime"
+
+	"github.com/hashicorp/vault/vault"
+)
+
+// handlePprofNamedIndexRequest returns an http.Handler that serves the pprof
+// profile corresponding to "name".  The list of valid profiles includes:
+// "cmdline", "profile", "symbol", "trace", "allocs", "block", "goroutine",
+// "heap", "mutex", "threadcreate".
+//
+// Where applicable, we apply runtime settings to ensure the profiles produce
+// output.
+func handlePprofNamedIndexRequest(core *vault.Core, name string) http.Handler {
```

Comment:
Might want to come up with a better name and get away from passing the entire core in here -- we only need the logger.

```suggestion
func handlePprofTargetRequest(logger hclog.Logger, name string) http.Handler {
```